src/schedulers/dpmsolver_multistep.rs

```diff
@@ -0,0 +1,368 @@
+                DPMSolverType::Midpoint => {
+                    (alpha_t / alpha_s0) * sample
+                        - (sigma_t * (h.exp() - 1.0)) * d0
+                        - 0.5 * (sigma_t * (h.exp() - 1.0)) * d1
+                }
+                DPMSolverType::Heun => {
+                    (alpha_t / alpha_s0) * sample
+                        - (sigma_t * (h.exp() - 1.0)) * d0
+                        - (sigma_t * ((h.exp() - 1.0) / h - 1.0)) * d1
+                }
+            },
+        }
+    }
+
+    /// One step for the third-order multistep DPM-Solver
+    fn multistep_dpm_solver_third_order_update(
+        &self,
+        model_output_list: &Vec<Tensor>,
+        timestep_list: [usize; 3],
+        prev_timestep: usize,
+        sample: &Tensor,
+    ) -> Tensor {
+        let (t, s0, s1, s2) = (
+            prev_timestep,
+            timestep_list[timestep_list.len() - 1],
+            timestep_list[timestep_list.len() - 2],
+            timestep_list[timestep_list.len() - 3],
+        );
+        let (m0, m1, m2) = (
+            model_output_list[model_output_list.len() - 1].as_ref(),
+            model_output_list[model_output_list.len() - 2].as_ref(),
+            model_output_list[model_output_list.len() - 3].as_ref(),
+        );
+        let (lambda_t, lambda_s0, lambda_s1, lambda_s2) =
+            (self.lambda_t[t], self.lambda_t[s0], self.lambda_t[s1], self.lambda_t[s2]);
+        let (alpha_t, alpha_s0) = (self.alpha_t[t], self.alpha_t[s0]);
+        let (sigma_t, sigma_s0) = (self.sigma_t[t], self.sigma_t[s0]);
+        let (h, h_0, h_1) = (lambda_t - lambda_s0, lambda_s0 - lambda_s1, lambda_s1 - lambda_s2);
+        let (r0, r1) = (h_0 / h, h_1 / h);
+        let d0 = m0;
+        let (d1_0, d1_1) = ((1.0 / r0) * (m0 - m1), (1.0 / r1) * (m1 - m2));
+        let d1 = &d1_0 + (r0 / (r0 + r1)) * (&d1_0 - &d1_1);
+        let d2 = (1.0 / (r0 + r1)) * (d1_0 - d1_1);
+
+        match self.config.algorithm_type {
+            DPMSolverAlgorithmType::DPMSolverPlusPlus => {
+                // See https://arxiv.org/abs/2206.00927 for detailed derivations
+                (sigma_t / sigma_s0) * sample - (alpha_t * ((-h).exp() - 1.0)) * d0
+                    + (alpha_t * (((-h).exp() - 1.0) / h + 1.0)) * d1
+                    - (alpha_t * (((-h).exp() - 1.0 + h) / h.powi(2) - 0.5)) * d2
+            }
+            DPMSolverAlgorithmType::DPMSolver => {
+                (alpha_t / alpha_s0) * sample
+                    - (sigma_t * (h.exp() - 1.0)) * d0
+                    - (sigma_t * ((h.exp() - 1.0) / h - 1.0)) * d1
+                    - (sigma_t * ((h.exp() - 1.0 - h) / h.powi(2) - 0.5)) * d2
+            }
+        }
+    }
+
+    pub fn timesteps(&self) -> &[usize] {
+        self.timesteps.as_slice()
+    }
+
+    pub fn step(&mut self, model_output: &Tensor, timestep: usize, sample: &Tensor) -> Tensor {
+        #[rustfmt::skip]
```

Comment:
Very cosmetic but could you avoid overriding the rustfmt setting? It's never the case in this or the tch crate and even if it might look weird at some point (and I would personally dislike some formatting), it makes things much easier if someone else edits the code.

Reply:
Sure, I did it to leave the chain of calls on multiple lines, otherwise rustfmt would force me to put everything on one line. If you don't like it, I will fix this in the next commit :)

Reply:
Well I really don't mind the format one way or another, I'm just opiniated towards avoiding manual formatting and so going to whatever rustfmt decides (and again I might certainly dislike some of the outputs but it's great not to have to think about manual formatting anymore :) ).

Reply:
Is there any way to keep this format without the action failing ? 
Otherwise I will just accept the format enforced by rustfmt, it ain't a big deal after all :)

Reply:
Yeah I would just go with what rusfmt suggests.

Reply:
Sure :)

Reply:
Sorry, maybe I missed something but the rustfmt skip seems to still be there? (once it has been removed/reformatted, happy to merge)

Reply:
I just committed :)